src/js/graph/node.js

```diff
@@ -67,69 +67,70 @@ Node.prototype.setGraph = function(graph) {
 };
 
 // Span/SpanContainer trait functions
-Node.prototype.getText = function() { 
+Node.prototype.getText = function() {
     var isSpan = this.hasTraitSpan(), isSpanContainer = this.hasTraitSpanContainer();
-    if (!isSpan && !isSpanContainer) { 
+    if (!isSpan && !isSpanContainer) {
         throw Error("Calling `getText` on a Node that does not have the `span` or `spanContainer` trait.");
     }
     return this._graph.getContent().slice(this.getStartIndex(), this.getEndIndex());
 };
-Node.prototype.getStartIndex = function() {
-    var isSpan = this.hasTraitSpan(), isSpanContainer = this.hasTraitSpanContainer();
-    if (!isSpan && !isSpanContainer) { 
-        throw Error("Calling `getStartIndex` on a Node that does not have the `span` or `spanContainer` trait.");
-    }
-    if (isSpan) { return this.getProp('start'); }
-    // Traverse the first edges until the nodes are no longer span containers.
-    var firstNode = this.getFirst();
-    while (true) {
-        var firstNodeIsSC = firstNode.hasTraitSpanContainer();
-        if (!firstNodeIsSC) { break; }
-        if (firstNodeIsSC) { firstNode = firstNode.getFirst(); }
+Node.prototype._getIndex = function(isStart) {
+    if (this.hasTraitSpan()) {
+        var start = this.getProp('start');
+        return isStart ? start : start + this.getProp('length');
     }
-    if (!firstNode.hasTraitSpan()) {
-        throw Error("Calling `getText` on a node [" + this._id + "] whose first edge does not resolve to a span.");
+    if (this.hasTraitSpanContainer()) {
+        return isStart ? this.getFirst().getStartIndex() : this.getLast().getEndIndex();
+    } else {
+        // Find all edges that are not the same type as the current node.
+        // And determine which one has the proper index.
+        var self = this;
+        var reduced = this.getEdges().filter(function(edge) {
+            return edge.getTargetType() !== self.getType();
+        }).reduce(function(idx, edge) {
+            var node = self._graph.getNodeById(edge.getTargetId());
+            var newIndex = isStart ? node.getStartIndex() : node.getEndIndex();
+            if (isStart) {
+                if (newIndex < idx) { return newIndex; }
+            } else {
+                if (idx < newIndex) { return newIndex; }
+            }
+            return idx;
+        }, isStart ? Infinity : Number.NEGATIVE_INFINITY);
```

Comment:
They are used to compare the max values vs the values of the start|end indices. The negative infinity isn't that useful, it could easily be -1. But the `Infinity` is useful since that's what's used to pull the end index down to the proper value.